hudi-spark-datasource/hudi-spark3.4.x/src/main/scala/org/apache/spark/sql/adapter/Spark3_4Adapter.scala

```diff
@@ -154,6 +154,13 @@ class Spark3_4Adapter extends BaseSpark3Adapter {
     Spark34OrcReader.build(vectorized, sqlConf, options, hadoopConf, dataSchema)
   }
 
+  override def createLanceFileReader(vectorized: Boolean,
+                                     sqlConf: SQLConf,
+                                     options: Map[String, String],
+                                     hadoopConf: Configuration): SparkColumnarFileReader = {
+    throw new UnsupportedOperationException("Lance format is not supported in Spark 3.4")
```

Comment:
Why can't we support lance with these lower spark versions?

Reply:
When I discussed with @yihua I believe we only wanted to support lance for the latest two versions in hudi spark(3.5, 4.0). 

In terms of what lance supports from the docs there is support for 3.4 and beyond https://lance.org/integrations/spark/install/#scala, so if we feel that we should include spark 3.4.0 then will add this.
<img width="1066" height="456" alt="Screenshot 2025-12-29 at 8 20 01 AM" src="https://github.com/user-attachments/assets/aa23910e-3f9f-4afe-9709-1e4ab8bbf04b" />

cc @nsivabalan

Reply:
If it is not a lot of extra work to add 3.4 support, then I don't see why we would skip it.

Reply:
I couldn't recall when the decision was made.  If it's easy to add Lance support on Spark 3.4, we should do that.